observer.go

```diff
@@ -24,7 +24,7 @@ import opentracing "github.com/opentracing/opentracing-go"
 
 // Observer can be registered with the Tracer to receive notifications about new Spans.
 type Observer interface {
-	OnStartSpan(operationName string, options opentracing.StartSpanOptions) SpanObserver
+	OnStartSpan(sp opentracing.Span, operationName string, options opentracing.StartSpanOptions) SpanObserver
```

Comment:
There's a problem with changing this interface - it is a breaking change and would require a major version upgrade. Given https://github.com/opentracing-contrib/go-observer/pull/1, what if we leave this unchanged (document as deprecated) and instead add support for "options.ContribObserver()"?  

We will support both until the next major release of jaeger-client, at which point we can consolidate into one AP.